src/mono/mono/mini/intrinsics.c

```diff
@@ -767,10 +767,10 @@ get_class_from_ldtoken_ins (MonoInst *ins)
  * their relation (EQ/NE/NONE).
  */
 static CompRelation
-get_rttype_ins_relation (MonoInst *ins1, MonoInst *ins2, gboolean *vtype_constrained_gparam)
+get_rttype_ins_relation (MonoCompile *cfg, MonoInst *ins1, MonoInst *ins2, gboolean *vtype_constrained_gparam)
 {
-	MonoClass *k1 = get_class_from_ldtoken_ins (ins1);
-	MonoClass *k2 = get_class_from_ldtoken_ins (ins2);
+	MonoClass *k1 = get_class_from_ldtoken_ins (cfg, ins1);
```

Comment:
Is this ever going to inflate a gshared type and make the `mono_class_is_gparam (k1)` case fail more often?

Reply:
gshared types are encoded using a different opcode (OP_RTTYPE).